src/FakeItEasy/Expressions/ParsedCallExpression.cs

```diff
@@ -34,5 +34,28 @@ internal class ParsedCallExpression
         public IEnumerable<ParsedArgumentExpression> ArgumentsExpressions { get; }
 
         public object CallTarget => this.callTarget.Value;
+
+        // Expression evaluation optimized for a very common case: 
+        // - it's a 'MemberAccess' expression on a local lambda closure (A.CallTo(() => fakeObj.doSomething(args));

```

Comment:
I enjoyed this comment. It helps me understand things now and I'm sure will help me in the future. 
I'm not quite there, though.
I _think_ the second case tells us that we're going after some member on the fake and then the first case identifies the actual member. Is that right?